src/Azure.Functions.Cli/Kubernetes/KEDA/KedaResource.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+namespace Azure.Functions.Cli.Kubernetes.KEDA
+{
+    public class KedaResource
+    {
+        public virtual IDictionary<string, string> PopulateMetadataDictionary(JToken t)
+        {
+            IDictionary<string, string> metadata = t.ToObject<Dictionary<string, JToken>>()
+                .Where(i => i.Value.Type == JTokenType.String)
+                .ToDictionary(k => k.Key, v => v.Value.ToString());
+
+            // TODO: Check for breaking changes
+            if (t["type"].ToString().Equals("rabbitMQTrigger", StringComparison.InvariantCultureIgnoreCase))
+            {
+                metadata["host"] = metadata["connectionStringSetting"];
+                metadata.Remove("connectionStringSetting");
+            }
```

Comment:
`connectionStringSetting` is not necessarily always present in case of `rabbitMQTrigger`. Sometimes they can choose to authenticate with `HostName`, `UserName`, `Password`.

In cases where both `connectionStringSetting`, `HostName`, `UserName`, `password` are all present, `connectionStringSetting` takes precedence and rest are ignored.

reference link: https://github.com/Azure/azure-functions-rabbitmq-extension/blob/dev/src/Services/RabbitMQService.cs

Reply:
I'll take a look at this tomorrow, thanks for suggesting!

Reply:
I'm afraid we won't be able to support that scenario as per https://keda.sh/docs/2.0/scalers/rabbitmq-queue/ unless @ahmelsayed has a good idea how to approach this?